### DISCUSSION TITLE
How to use onChange with Controller?

### DISCUSSION THREAD

**Discussion by YolkKanapon** [react-hook-form/react-hook-form#4372] [Q&A] [answered]
`<Controller
            name="firstName"
            as={
              <InputBox
                variant="outlined"
                id="firstName"
                label={en.pages.firstName}
                value={values.firstName}
                helperText={
                  fieldsErrors.firstName ? fieldsErrors.firstName.message : null
                }
                onChange={handleInputChange}
                errors={fieldsErrors.firstName}
              />
            }
            control={control}
            defaultValue=""
            rules={{
              required: en.validate.fieldRequire,
            }}
          />`
Can someone help me? I try to use custom onChange to setState but it doesn't work.

**Comment by bluebill1049** [comment, marked as answer]
use `render` prop instead.

**Reply by YolkKanapon** [reply]
Thanks.
